Controller/FolderController.php

```diff
@@ -0,0 +1,162 @@
+<?php
+/**
+ * File containing the FolderController class.
+ *
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ * @version //autogentag//
+ */
+
+namespace EzSystems\DemoBundle\Controller;
+
+use eZ\Publish\API\Repository\Values\Content\LocationQuery;
+use eZ\Bundle\EzPublishCoreBundle\Controller;
+use eZ\Publish\Core\Pagination\Pagerfanta\ContentSearchAdapter;
+use Pagerfanta\Pagerfanta;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use eZ\Publish\API\Repository\Values\Content\Query;
+use eZ\Publish\API\Repository\Values\Content\Query\SortClause;
+
+class FolderController extends Controller
+{
+    /**
+     * Displays the sub folder if it exists
+     *
+     * @param int $locationId of a folder
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function showFolderListAsideViewAction( $locationId )
```

Comment:
Since @yannickroger has merged the ParamConverter PR, you can directly typehint the Location object, and it will be loaded for you:

```
public function showFolderListAsideViewAction( Location $location )
```
And of course, you can remiove the getRepository, getLocationService and loadLocation calls. below (but keep the invisible check).

The same improvement can be applied to all controller actions, as long as the argument is a Location or a Content.